lib/puppet/provider/firewall/iptables.rb

```diff
@@ -492,6 +494,8 @@ def self.rule_to_hash(line, table, counter)
                  values.sub(%r{\s-f(?!l)(?=.*--comment)}, ' -f true')
                elsif resource_map[bool].eql?(%r{'--physdev-is-\S+'})
                  values.sub(%r{'#{resource_map[bool]} "! "'}, "#{resource_map[bool]} true")
+               elsif bool == :random
+                 values.sub(%r{'#{resource_map[bool]}(\s|$)'}, "#{resource_map[bool]} true")
```

Comment:
Based on the failing tests it looks like this needs to incorporate the same logic for negation that the else case handles.

I think use of `\b` (https://www.regular-expressions.info/wordboundaries.html) will allow this to just be a patch on the else case instead of making `:random` a special case. 

```suggestion
               else
                 # append `true` to booleans that are not already negated (followed by "!")
                 values.sub(%r{'#{resource_map[bool]}\b(?! "!")'}, "#{resource_map[bool]} true")
```